From 352732967a05dc51f1a769b632a1dff996ce2ef5 Mon Sep 17 00:00:00 2001
From: James Prestwood <james.prestwood@linux.intel.com>
Date: Tue, 6 Aug 2019 13:46:28 -0700
Subject: dhcp: fix T1 timeout crash

The timeout user data was being passed as the address to the
client object, not the client object itself.
---
 ell/dhcp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ell/dhcp.c b/ell/dhcp.c
index ca81311..22d903e 100644
--- a/ell/dhcp.c
+++ b/ell/dhcp.c
@@ -985,7 +985,7 @@ static void dhcp_client_rx_message(const void *data, size_t len, void *userdata)
 		client->timeout_lease =
 			l_timeout_create_ms(dhcp_fuzz_secs(client->lease->t1),
 						dhcp_client_t1_expired,
-						&client, NULL);
+						client, NULL);
 
 		break;
 	case DHCP_STATE_INIT_REBOOT:
-- 
cgit 1.2-0.3.lf.el7

